Lib/fontTools/varLib/errors.py

```diff
@@ -45,67 +41,121 @@ def _master_name(self, ix):
         else:
             return "master number %i" % ix
 
-    def _offender(self, cause):
-        reason = cause["reason"].value
-        if "expected" in cause and "got" in cause:
-            index = [x == cause["expected"] for x in cause["got"]].index(False)
-            return index, self._master_name(index)
-        if reason == VarLibMergeFailure.FoundANone:
-            index = [x is None for x in cause["got"]].index(True)
+    @property
+    def offender(self):
+        if "expected" in self.cause and "got" in self.cause:
+            index = [x == self.cause["expected"] for x in self.cause["got"]].index(
+                False
+            )
             return index, self._master_name(index)
         return None, None
 
-    def _incompatible_features(self, offender_index):
-        cause, stack = self.args[0], self.args[1:]
+    @property
+    def details(self):
+        if "expected" in self.cause and "got" in self.cause:
+            offender_index, offender = self.offender
+            got = self.cause["got"][offender_index]
+            return f"Expected to see {self.stack[0]}=={self.cause['expected']}, instead saw {got}\n"
+        return ""
+
+    def __str__(self):
+        offender_index, offender = self.offender
+        location = ""
+        if offender:
+            location = f"\n\nThe problem is likely to be in {offender}:\n"
+        context = "".join(reversed(self.stack))
+        basic = textwrap.fill(
+            f"Couldn't merge the fonts, because {self.reason}. "
+            f"This happened while performing the following operation: {context}",
+            width=78,
+        )
+        return "\n\n" + basic + location + self.details
+
+
+class VarLibMergeFailureShouldBeConstant(VarLibMergeError):
+    """some values were different, but should have been the same"""
+
+    @property
+    def details(self):
+        if self.stack[0] != ".FeatureCount":
+            return super()
```

Comment:
I think this should be:

```suggestion
            return super().details
```

or maybe you're right and I learned a new trick?  worth testing this